corehq/apps/userreports/reports/data_source.py

```diff
@@ -61,7 +66,14 @@ def backend(self):
     @property
     def data_source(self):
         if self._data_source is None:
-            if self.backend == UCR_ES_BACKEND:
+            if self._custom_query_provider:
+                self._data_source = ConfigurableReportCustomDataSource(
+                    self.domain, self.config, self._filters,
+                    self._aggregation_columns, self._columns,
+                    self._order_by
+                )
+                self._data_source.set_provider(self._custom_query_provider)
```

Comment:
any reason not to set this in the constructor?

Reply:
Yeah. I think that's why that was necessary